### PR TITLE
Refactoring 1_stdlib/Optional.swift test file

### PR DESCRIPTION
**Pull request by frootloops**
Can somebody have a look at the PR and leave some comments and advices about code style, variable naming etc.

I use `./llvm/utils/lit/lit.py -sv ./build/Ninja-DebugAssert/swift-macosx-x86_64/test-macosx-x86_64/1_stdlib/Optional.swift` for running tests. Is it right way to do so?

### CODE REVIEW

File: test/1_stdlib/Optional.swift [hunk 1]
```diff
@@ -1,226 +1,153 @@
...
+import StdlibUnittest
+import Swift
+
+let OptionalTests = TestSuite("Optional")
+
 protocol TestProtocol1 {}
 
 // Check that the generic parameter is called 'Memory'.
-extension Optional where Wrapped : TestProtocol1 {
+extension Optional where Wrapped: TestProtocol1 {
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: In the standard library code, we put spaces on both sides of the colon when it denotes an "is-a" relationship: in generic parameter lists, in inheritance and protocol implementation lists etc.

File: test/1_stdlib/Optional.swift [hunk 2]
```diff
@@ -1,226 +1,153 @@
...
-}
+OptionalTests.test("nil comparison") {
+  var x: Int? = nil
+  expectFalse(x != nil)
 
-x = .Some(0)
-
-x = .Some(1)
+  switch x {
+  case .Some(let y): assert(false)
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: expectUnreachable()

File: test/1_stdlib/Optional.swift [hunk 3]
```diff
@@ -1,226 +1,153 @@
...
+  case .None: assert(true)
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: assert => `break`

**Comment by frootloops** [comment]
@gribozavr thanks for review. Can you look again :)

**Comment by frootloops** [comment]
Whats about `Optional<Int>` vs `Int?` What do you prefer?

**Comment by gribozavr** [comment]
`Int?`, where parser accepts it.

**Comment by frootloops** [comment]
@gribozavr `Optional<Int>` and `Int?` are they equal?

**Comment by gribozavr** [comment]
@frootloops Yes, `T?` is syntax sugar for `Optional<T>`.

**Comment by frootloops** [comment]
@gribozavr ok, thanks.

**Comment by frootloops** [comment]
@gribozavr is PR ready for merge?

File: test/1_stdlib/Optional.swift [hunk 4]
```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
...
-// CHECK-NEXT: true, false, false, false, false, true.
-
-testRelation(!=)
-// CHECK-NEXT: false, true, true, true, true, false
-
-testRelation(<)
-// CHECK-NEXT: false, true, false, false, true, false.
+expectEqual(testRelation(==), [true, false, false, false, false, true])
+expectEqual(testRelation(!=), [false, true, true, true, true, false])
+expectEqual(testRelation(<),  [false, true, false, false, true, false])
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: Please move this into an `OptionalTest.test() {}`, there should be no top-level code.

File: test/1_stdlib/Optional.swift [hunk 5]
```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
...
 
 struct X {}
 class C {}
 
-class E : Equatable {}
+class E: Equatable {}
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: Space after the colon, please.

File: test/1_stdlib/Optional.swift [hunk 6]
```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
...
+  func nextCounter2() -> Int? { return nextCounter() }
+
+  let a: Int? = 123
+  let b: Int? = nil
+  let c: Int? = nil
+  let d: Int? = 456
+  let e: Int? = nil
+  let f: Int? = nil
+
+  expectEqual(a ?? nextCounter(), 123)
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: `expectEqual()` accepts the expected value as the first parameter (it matters in the error message when the test fails).

File: test/1_stdlib/Optional.swift [hunk 7]
```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
...
 struct X {}
 class C {}
 
-class E : Equatable {}
+class E: Equatable {}
 func == (_: E, _: E) -> Bool { return true }
 
-func nilComparison() {
-  let _: X? = nil
-  let _: X? = X()
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: Please move these two lines into a separate test.

File: test/1_stdlib/Optional.swift [hunk 8]
```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
...
-  print(v0 == nil) // CHECK-NEXT: true
-  print(v0 != nil) // CHECK-NEXT: false
-
-  print(nil == v1) // CHECK-NEXT: false
-  print(nil != v1) // CHECK-NEXT: true
-  print(nil == v0) // CHECK-NEXT: true
-  print(nil != v0) // CHECK-NEXT: false
-
-  let _: C? = nil
-  let _: C? = C()
```
Comment (gribozavr) [inline test/1_stdlib/Optional.swift]: Please move these two lines into a separate test.

**Comment by frootloops** [comment]
@gribozavr I have update the PR.

**Comment by gribozavr** [comment]
@frootloops Thank you!

**Pull request merged**